HLTriggerOffline/Common/python/customizeHLTforAlpakavsCUDA.py

```diff
@@ -0,0 +1,911 @@
+        src = cms.InputTag('hltSiPixelDigisSoACUDA'),
+        clusterThreshold_layer1 = cms.int32(4000),
+        clusterThreshold_otherLayers = cms.int32(4000),
+        produceDigis = cms.bool(False),
+        storeDigis = cms.bool(False)
+    )
+
+    # Not sure if clusters cache are needed for this comparison
+    process.hltSiPixelClustersCacheCUDA = cms.EDProducer('SiPixelClusterShapeCacheProducer',
+        src = cms.InputTag( 'hltSiPixelClustersCUDA' ),
+        onDemand = cms.bool( False )
+    )
+
+    # legacy EDProducer
+    # consumes
+    #  - SiPixelDigiErrorsHost
+    #  - SiPixelFormatterErrors
+    # produces
+    #  - edm::DetSetVector<SiPixelRawDataError>
+    #  - DetIdCollection
+    #  - DetIdCollection, 'UserErrorModules'
+    #  - edmNew::DetSetVector<PixelFEDChannel>
+    process.hltSiPixelDigisCUDA = cms.EDProducer('SiPixelDigiErrorsFromSoA',
+        digiErrorSoASrc = cms.InputTag('hltSiPixelDigisErrorsSoACUDA'),
+        CablingMapLabel = cms.string(''),
+        UsePhase1 = cms.bool(True),
+        ErrorList = cms.vint32(29),
+        UserErrorList = cms.vint32(40)
+    )
+
+    # CUDA EDProducer
+    # consumes
+    #  - BeamSpotCUDA
+    #  - SiPixelClustersCUDA
+    #  - SiPixelDigisCUDA
+    # produces
+    #  - TrackingRecHitSoADevice<TrackerTraits>
+    process.hltSiPixelRecHitsSoACUDA = cms.EDProducer("SiPixelRecHitCUDAPhase1",
+        CPE = cms.string('PixelCPEFast'),
+        beamSpot = cms.InputTag("hltOnlineBeamSpotGPU"),
+        mightGet = cms.optional.untracked.vstring,
+        src = cms.InputTag("hltSiPixelClustersSoACUDA")
+    )
+
+    # legacy EDProducer
+    # consumes
+    #  - TrackingRecHitSoADevice<TrackerTraits>
+    #  - SiPixelClusterCollectionNew
+    # produces
+    #  - SiPixelRecHitCollection
+    #  - HMSstorage
+    process.hltSiPixelRecHitsCUDA = cms.EDProducer('SiPixelRecHitFromCUDAPhase1',
+        pixelRecHitSrc = cms.InputTag('hltSiPixelRecHitsSoACUDA'),
+        src = cms.InputTag('hltSiPixelClustersCUDA'),
+    )
+
+    # Needed to convert hits on device to hits on host for monitoring/comparisons
+    # CUDA EDProducer
+    # consumes
+    #  - TrackingRecHitSoADevice<TrackerTraits>
+    # produces
+    #  - TrackingRecHitSoAHost<TrackerTraits>
+    #  - HMSstorage
+    process.hltSiPixelRecHitsSoAFromCUDA = cms.EDProducer('SiPixelRecHitSoAFromCUDAPhase1',
+        pixelRecHitSrc = cms.InputTag('hltSiPixelRecHitsSoACUDA'),                                             
+    )
+
+    ###
+    ### Task: Pixel Local Reconstruction
+    ###
+    process.HLTDoLocalPixelCUDATask = cms.ConditionalTask(
+        process.hltOnlineBeamSpotGPU,
+        process.hltSiPixelClustersSoACUDA,
+        process.hltSiPixelDigisSoACUDA,
+        process.hltSiPixelDigisErrorsSoACUDA,
+        process.hltSiPixelClustersCUDA,   # was: hltSiPixelClusters
+        process.hltSiPixelClustersCacheCUDA,          # really needed ??
```

Comment:
have you checked?

Reply:
Running the comparisons with or without it doesn't make any difference (at least at step 2 level). I don't see it being used as input to any other module or as a collection in the output root file (this is almost a copy of https://github.com/cms-sw/cmssw/blob/master/HLTrigger/Configuration/python/customizeHLTforAlpaka.py, so the comment came from there). I don't think this is useful for the alpaka serial vs device or alpaka vs cuda comparisons (again, at least at step 2 level), as it is not used further. Maybe it can be removed from the HLT customizations, then?

Reply:
@mmusich do you have any comment/suggestion about this?

Reply:
>  I don't think this is useful for the alpaka serial vs device or alpaka vs cuda comparisons (again, at least at step 2 level), as it is not used further. Maybe it can be removed from the HLT customizations, then?

if it's really not used, it's indeed better to remove it.

Reply:
@mmusich so, I took a deeper look at this. Even though for the comparison itself (the DQM histograms of Alpaka vs CUDA objects) this plugin doesn't has any influence, it is produced as an output of most `HLT` and/or `DEBUG` event contents. I assume it would be reasonable to keep it for these comparisons, specially since this should be used for validation purposes. Please, let me know if you have any further considerations about this

Reply:
@mmusich could you please comment on my point above?

Reply:
I *think* it should be removed.

Reply:
Alright. I'll remove them for the comparison. Thanks!